cips/cip-18.md

```diff
@@ -38,49 +38,62 @@ The following API is proposed for the standardised gas estimation service.
 
 ```proto
 service GasEstimator {
-    rpc EstimateGasUsage(EstimateGasUsageRequest) returns (EstimateGasUsageResponse) {}
-    rpc EstimateGasPrice(EstimateGasPriceRequest) returns (EstimateGasPriceResponse) {}
+    rpc EstimateFee(EstimateFeeRequest) returns (EstimateFeeResponse) {}
+    rpc EstimateFeeAndGas(EstimateFeeAndGasRequest) returns (EstimateFeeAndGasResponse) {}
 }
 
-message EstimateGasUsageRequest {
-    cosmos.tx.Tx tx = 1;
+enum TxPriority {
+  NONE = 0;
+  LOW = 1;
+  MEDIUM = 2;
+  HIGH = 3;
 }
 
-message EstimateGasUsageResponse {
-    uint64 estimated_gas_used = 1;
+message EstimateFeeRequest {
+    TxPriority tx_priority = 1;
 }
 
-message EstimateGasPriceRequest {}
+message EstimateFeeResponse {
+    double estimated_gas_price = 1;
+}
```

Comment:
[nit] I liked the previous names b/c `fee` implies gas price * gas and this response doesn't include an `estimated_gas` field

Reply:
I had changed it to `fee` for brevity reasons but I got the same feeling around the accuracy of names as you. Maybe I will change it back and have it longer